### PR TITLE
Add enpoints

### PR DESCRIPTION
**Pull request by Thompadude**
Add relations endpoints. This change will enable the client to handle relations between components and external sources.

### CODE REVIEW

File: rest/client.go [hunk 1]
```diff
@@ -49,3 +54,63 @@ func (c *client) GetComponentsByAsset(ctx context.Context, id uuid.UUID, filter
 
 	return response.Components, nil
 }
+
+func (c *client) GetComponentRelations(ctx context.Context, id uuid.UUID, limit int, continuationToken string) (models.GetComponentRelationsResponse, error) {
```
Comment (Ekberg) [inline rest/client.go]: We usually do not expose the paging operation in the clients.

So GetComponentRelations/GetRelatiedComponents should probably just return all and use the pagination if needed. This avoids everyone needing to write the pagination logic on their end.

File: rest/client.go [hunk 2]
```diff
@@ -49,3 +54,63 @@ func (c *client) GetComponentsByAsset(ctx context.Context, id uuid.UUID, filter
...
+		SetHeader("Accept", "application/json")
+
+	if _, err := c.Do(ctx, request); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *client) DeleteComponentRelation(ctx context.Context, externalID, componentID uuid.UUID, source, relationType string) (err error) {
```
Comment (Ekberg) [inline rest/client.go]: This one should probably take a `models.Relation` as an argument instead of splitting the relation as arguments

Comment (Thompadude) [reply rest/client.go]: Fixed!

File: rest/client.go (hunk 1 above)
Comment (Thompadude) [reply rest/client.go]: Makes sense - updated.

File: rest/client.go [hunk 3]
```diff
@@ -16,6 +18,13 @@ type GetComponentsFilter struct {
 
 type Client interface {
 	GetComponentsByAsset(context.Context, uuid.UUID, GetComponentsFilter) ([]models.Component, error)
+
+	GetComponentRelations(context.Context, uuid.UUID) ([]models.Relation, error)
+	GetComponentRelationsPage(context.Context, uuid.UUID, int, string) (models.GetComponentRelationsResponse, error)
```
Comment (Ekberg) [inline rest/client.go]: I'm happy to merge this if we just make the ...Page functions non-public. Most consumers should not need them.

Comment (Thompadude) [reply rest/client.go]: Makes sense! :) Updated.

**Review by Ekberg — APPROVED** [review]

**Pull request merged**